client/src/components/Search.jsx

```diff
@@ -1,79 +1,166 @@
 import React from "react";
 import ReactDOM from "react-dom";
+import axios from "axios";
 
 class Search extends React.Component {
 	constructor(props) {
 		super(props);
-		this.state = {};
+		this.state = {
+			location: "",
+			checkIn: "",
+			checkOut: "",
+			guests: "",
+			options: []
+		};
 	}
 
+	handleChangeLocation(e) {
+		this.setState({
+			location: e.target.value
+		});
+	}
+
+	handleChangeCheckIn(e) {
+		this.setState({
+			checkIn: e.target.value
+		});
+	}
+
+	handleChangeCheckOut(e) {
+		this.setState({
+			checkOut: e.target.value
+		});
+	}
+
+	chooseGuest(e) {
+		this.setState({
+			guests: e.target.value
+		});
+	}
+
+	getHostelOptions() {
+		axios
+			.get("/locations/hostels")
+			.then(response => {
+				const options = [];
+				console.log("get hostel options", response.data);
+				for (let i = 0; i < response.data.length; i += 1) {
+					if (response.data[i].city === this.state.location) {
+						options.push(response.data[i]);
+					}
+				}
+				this.setState({
+					options: options
+				});
+
+				if (this.state.location === "" && options.length === 0) {
```

Comment:
Great job on handling unfound searches! You might want to consider moving the conditionals in a helper function and then calling the helper function. This will make your getHostelOptions function look cleaner.